Avoid whole-form validation when no field validation is provided.

lucasconstantino opened the pull request:
Having a default validation function provided - even when no field validator is given - results in whole form validation being triggered each time array items are altered:

https://github.com/final-form/final-form/blob/9c732e27f29e82ea55b7c0f8a6cafb208c96ccb9/src/FinalForm.js#L897

This PR fixes it, by ensuring `undefined` is passed in case no validation is provided by the user.

<!--

👋 Hey, thanks for your interest in contributing to 🏁 React Final Form Arrays!

**Please ask first before starting work on any significant new features.**

It's never a fun experience to have your pull request declined after investing a
lot of time and effort into a new feature. To avoid this from happening, we
request that contributors create an issue to first discuss any significant new
features.

Please try to keep your pull request focused in scope and avoid including
unrelated commits.

After you have submitted your pull request, we’ll try to get back to you as soon
as possible. We may suggest some changes or improvements.

Please format the code before submitting your pull request by running:

```
npm run precommit
```

https://github.com/final-form/react-final-form-arrays/blob/master/.github/CONTRIBUTING.md

-->

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive test cases for form field array validation to ensure validators are properly executed when configured and correctly handled when validation is not configured.

* **Refactor**
  * Refactored the field array validation system to improve how validators are conditionally initialized and executed, with corresponding enhancements to error handling during field operations.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

lucasconstantino committed:
feat: ensure empty validation doesn't force form validation

Commit:
Add tests for validation optimization

- Test verifies no validator is registered when validateProp is undefined
- Test verifies validator is called when validateProp is provided
- Tests ensure the fix prevents unnecessary form-wide validation

erikras-richard-agent reviewed (dismissed):
LGTM! 

This is a nice optimization that avoids triggering unnecessary form-wide validation when the field array doesn't have a validator. The implementation is clean:

✅ Conditionally returns `undefined` instead of a no-op function
✅ Comprehensive tests for both scenarios (with/without validator)
✅ No breaking changes - behavior is preserved when validator is provided

The tests properly verify that:
- Without validate prop: no validation overhead
- With validate prop: validation runs as expected on initial render and mutations

Ready to merge.

Commit:
Address CodeRabbit feedback

- Fix type annotation: validate should be FieldValidator | undefined
- Update test to explicitly verify validate is undefined via useField spy

erikras-dinesh-agent reviewed (approved):
✅ Approved - CodeRabbit feedback addressed. Type annotation fixed and test updated to explicitly verify validate is undefined.

erikras-richard-agent commented:
Published in [`v5.0.0`](https://github.com/final-form/react-final-form-arrays/releases/tag/v5.0.0)
